core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -253,6 +236,22 @@ public void commit() throws IOException {
         }
     }
 
+    private static void move(Path source, Path destination) throws IOException {
+        if (atomicMoveSupported) {
+            try {
+                // Try to make an atomic move.
+                Files.move(source, destination, StandardCopyOption.ATOMIC_MOVE);
```

Comment:
```suggestion
                Files.move(source, destination, StandardCopyOption.ATOMIC_MOVE);
                return;
```
to avoid a race condition if another thread just turned this flag off.